### PR TITLE
Windowed iteration over Iterables.

### LINKED ISSUE #17: Add methods for windowed iteration over sequences
I use these all the time, but for some reason they are not in the standard library (not even itertools)

### PR DESCRIPTION
**Pull request by rgabbard**
Closes #17

### CODE REVIEW

File: vistautils/iterutils.py [hunk 1]
```diff
@@ -1,11 +1,12 @@
-# why does itertools think these are useful enough to put in the documentation but not in
-# the library?
-from itertools import tee
+from typing import TypeVar, Iterable, Tuple
 
+from vistautils.iter_utils import windowed
```
Comment (ConstantineLignos) [inline vistautils/iterutils.py]: Is all usage of `iterutils` deprecated? If so, I think this import above may enable people to accidentally used `iterutils.windowed`. Not sure what to do about that unless you import it as a different name here or something.

File: tests/test_iter_utils.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+from vistautils.iter_utils import windowed
+
+
+class TestIterUtils(TestCase):
+    def test_windowed_bad_window_size(self):
+        # need positive window size
+        with self.assertRaises(ValueError):
+            windowed(range(5), 0)
+            windowed(range(5), 0)
```
Comment (ConstantineLignos) [inline tests/test_iter_utils.py]: Is the fact you call it twice here meaningful?

File: vistautils/iter_utils.py [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+
+    Implementation borrowed from Jerry Kindall and ShadowRange(?) at
+    https://stackoverflow.com/a/40937024/413345
+    """
+    it = iter(iterable)
+    win = deque(islice(it, window_size), window_size)
+    if len(win) < window_size:
+        return
+    # cache method access for slight speed boost
+    append = win.append
```
Comment (ConstantineLignos) [inline vistautils/iter_utils.py]: My favorite Python optimization!

File: vistautils/iterutils.py (hunk 1 above)
Comment (rgabbard) [reply vistautils/iterutils.py]: @ConstantineLignos : Well, `iterutils` had just that one function (which is a special case of `windowed`) and all our other module are named with an underscore, so I wanted to deprecate `iterutils` in favor of `iter_utils`.  I'll change the import to discourage abuse accessing `windowed` the wrong way

File: tests/test_iter_utils.py (hunk 2 above)
Comment (rgabbard) [reply tests/test_iter_utils.py]: Nope, just a cut-and-paste-o

**Pull request merged**